lib_nbgl/include/nbgl_use_case_v2.h

```diff
@@ -0,0 +1,162 @@
+    nbgl_contentCallback_t     contentGetterCallback;  ///< function to call to retrieve a given content
+    uint8_t                    nbContents;  ///< number of contents
+} nbgl_genericContents_t;
+
+
+/*
+ * Generic Nav, should be used only in non-nominal cases (for example sub-settings levels)
+ */
+
+// API to polish, but main idea is to specify an array of contents and a navigation configuration
+// like touchable, progressIndicator and all...
+// this should be used in apps:
+// - that were previously using nbgl_useCaseSettings for that
+// - that have custom content to display that doesn't fit classical cases?
+void nbgl_useCaseGenericNav(const char                   *title,
+                            const nbgl_genericContents_t *contents,
+                            uint8_t                       initPage,
+                            bool                          touchable,
+                            bool                          progressIndicator,
+                            nbgl_callback_t               quitCallback);
+
+
+/*
+ * Home and settings
+ */
+
+// TODO add something to be able to restart the useCase at a specific page (for example after a switch confirmation page)
+// Probably an initPage params, taking an arbitrary value that would have been given in the switch callback.
+void nbgl_useCaseHomeAndSettings(const char                   *appName,
+                                 const nbgl_icon_details_t    *appIcon,
+                                 const char                   *tagline,
+                                 const nbgl_contentInfoList_t *infosList,
+                                 const nbgl_genericContents_t *settingContents,
+                                 nbgl_callback_t               quitCallback)
+
+// TODO add support for plugin and HomeExt use cases.
+
+/*
+ * TX review, exist in three variants "transaction", "message" and "operation"
+ * Then all wording are handled by the SDK and not exposed to the user.
+ * This bundles nbgl_useCaseReviewStart / nbgl_useCaseStaticReview and nbgl_useCaseConfirm
+ */
+
+void nbgl_useCaseTransactionReview(
+    const nbgl_layoutTagValueList_t *tagValueList,
+    const nbgl_icon_details_t *icon,
+    const char *reviewTitle,
+    const char *reviewSubTitle, /* Most often this is empty, but sometimes indicates a path / index */
+    const char *finish_page_text, /* unused on Nano */
+    nbgl_choiceCallback_t choice_callback);
+
+void nbgl_useCaseMessageReview(
+    const nbgl_layoutTagValueList_t *tagValueList,
+    const nbgl_icon_details_t *icon,
+    const char *reviewTitle,
+    const char *reviewSubTitle, /* Most often this is empty, but sometimes indicates a path / index */
+    const char *finish_page_text, /* unused on Nano */
+    nbgl_choiceCallback_t choice_callback);
+
+void nbgl_useCaseOperationReview(
+    const nbgl_layoutTagValueList_t *tagValueList,
```

Comment:
For each case where `tagValueList` is provided, it would be usefull (required for _Favorite Accounts_ feature) to have the possibility to display an additional touchable icon, on the tag value line, allowing to get more details (this would open a new popup)

Reply:
There is already a `valueIcon` for `tagValueList` items. I guess at some point we will need to add a token and a callback or something...
Maybe it could comes with an `*Ext` version of the functions taking something else than a `nbgl_layoutTagValueList_t` to avoid adding to much fuzz in it for nothing in most cases.